public/app/features/templating/template_srv.ts

```diff
@@ -8,7 +8,7 @@ function luceneEscape(value) {
 export class TemplateSrv {
   variables: any[];
 
-  private regex = /\$(\w+)|\[\[([\s\S]+?)\]\]/g;
+  private regex = /\$(\w+)|\[\[([\s\S]+?)(?::(\w+))?\]\]|\${(\w+)(?::(\w+))?}/g;
```

Comment:
I wonder if there is a good way to make it a little easier to understand. Esp this regex. Would it make sense to splitt it into two regexp and try to use the new format first and fall back to the old regex?

Its not a blocker but I know we want to improve this feature so making it easier to work with would be great.

Reply:
Named capturing groups would be great, but JS doesn't support them.  About the only suggestions that come to mind are that we could build the RegExp up out of multiple strings, but then you have to double-escape the backslashes and the whole thing just gets more opaque.

A comment might not hurt here to explain what's going on, the other thing we could do would be to rename the `gX` parameters lower down to `valueX` and `formatX` which might help make things clearer.  I'll take a shot at that here and see what I can come up with.